### PR TITLE
Patch 6

### PR DESCRIPTION
**Pull request by lelilia**
Since we added required to the date in the notes module we need to have a date set in the database.

So I added dates to the initial database (because we create the notes with save and not send so the controller for post is not called that adds the date for us

I also added a bit of code to change the date from an object to a string in the model

### CODE REVIEW

File: src/content/4/en/part4b.md [hunk 1]
```diff
@@ -723,6 +727,19 @@ test('a note can be deleted', async () => {
...
 ```
+In the test for 'a specific note can be viewed' we get an error because like the ID in 3b the date is stored as an Object in Mongoose without extra formatting but is compared to a JSON string.
+
+Let's add this to our models/note.js
+
+```js
+noteSchema.set('toJSON', {
+  transform: (document, returnedObject) => {
+  returnedObject.id = returnedObject._id.toString()
+  returnedObject.date = returnedObject.date.toString() // highlight-line
```
Comment (Kaltsoon) [inline src/content/4/en/part4b.md:738]: I would rely on the `new Date().toJSON()` implementation here, which is `new Date().toISOString()`

Comment (lelilia) [reply src/content/4/en/part4b.md:738]: looks less hacky than my attempt at fixing it with the knowledge of the course material :)

**Comment by Kaltsoon** [comment]
The core issue is related to this piece of code:

```javascript
const notesAtStart = await helper.notesInDb()

const noteToView = notesAtStart[0]

const resultNote = await API
  .get(`/api/notes/${noteToView.id}`)
  .expect(200)
  .expect('Content-Type', /application\/json/)

expect(resultNote.body).toEqual(noteToView)
```

`resultNode.body` goes through JSON serialization and parsing whereas `noteToView` doesn't. This results in the date being a string in the `resultNode.body` and a `Date` object in the `noteToView`. The easiest way to fix this issue is to  change the code into:

```javascript
const processedNoteToView = JSON.parse(JSON.stringify(noteToView));

expect(resultNote.body).toEqual(processedNoteToView)
```

However, this requires an explanation.

**Comment by Kaltsoon** [comment]
@lelilia are you able to check if the solution I provided actually works? Including the dates, you added to the note objects.

**Comment by lelilia** [comment]
yes that does the trick \o/

**Comment by Kaltsoon** [comment]
Awesome, thanks a lot! This has caused a lot of headache

**Pull request merged**